pkg/controllers/cluster/taint_manager_test.go

```diff
@@ -0,0 +1,428 @@
+				},
+				Spec: workv1alpha2.ResourceBindingSpec{
+					Clusters: []workv1alpha2.TargetCluster{
+						{
+							Name:     "test-cluster",
+							Replicas: 1,
+						},
+					},
+				},
+			},
+			cluster: &clusterv1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-cluster",
+				},
+				Spec: clusterv1alpha1.ClusterSpec{
+					Taints: []corev1.Taint{
+						{
+							Key:    "cluster.karmada.io/not-ready",
+							Effect: corev1.TaintEffectNoExecute,
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "rb with tolerations",
+			rb: &workv1alpha2.ResourceBinding{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ResourceBinding",
+					APIVersion: "work.karmada.io/v1alpha2",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "test-rb",
+					Namespace:   "default",
+					Annotations: map[string]string{"policy.karmada.io/applied-placement": `{"clusterAffinity":{"clusterNames":["member1","member2"]},"clusterTolerations":[{"key":"cluster.karmada.io/not-ready","operator":"Exists","effect":"NoExecute","tolerationSeconds":30},{"key":"cluster.karmada.io/unreachable","operator":"Exists","effect":"NoExecute","tolerationSeconds":30}],"replicaScheduling":{"replicaSchedulingType":"Divided","replicaDivisionPreference":"Weighted","weightPreference":{"staticWeightList":[{"targetCluster":{"clusterNames":["member1"]},"weight":1},{"targetCluster":{"clusterNames":["member2"]},"weight":1}]}}}`},
+				},
+				Spec: workv1alpha2.ResourceBindingSpec{
+					Clusters: []workv1alpha2.TargetCluster{
+						{
+							Name:     "test-cluster",
+							Replicas: 1,
+						},
+					},
+				},
+			},
+			cluster: &clusterv1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-cluster",
+				},
+				Spec: clusterv1alpha1.ClusterSpec{
+					Taints: []corev1.Taint{
+						{
+							Key:    "cluster.karmada.io/not-ready",
+							Effect: corev1.TaintEffectNoExecute,
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "rb not exist",
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tc := newNoExecuteTaintManager()
+			if tt.rb != nil {
+				if err := tc.Create(context.Background(), tt.rb, &client.CreateOptions{}); err != nil {
+					t.Fatalf("failed to create rb: %v", err)
+				}
+			}
+
+			if tt.cluster != nil {
+				if err := tc.Create(context.Background(), tt.cluster, &client.CreateOptions{}); err != nil {
+					t.Fatalf("failed to create cluster: %v", err)
+				}
+			}
+
+			key := keys.FederatedKey{
+				Cluster: "test-cluster",
+				ClusterWideKey: keys.ClusterWideKey{
+					Kind:      "ResourceBinding",
+					Name:      "test-rb",
+					Namespace: "default",
+				},
+			}
+			if err := tc.syncBindingEviction(key); (err != nil) != tt.wantErr {
```

Comment:
It's not enough to judge err.

Can you move the addition of taint_manager_test.go to the next PR?

Reply:
let me add some detail for this test case

Reply:
hi, i have added some details about this,  could you check it ?  if it still not enough, i will consider your suggestion and moving the taint_manager_test.go to the next pr.